validator/client/src/test/java/tech/pegasys/teku/validator/client/VoluntaryExitDataProviderTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright ConsenSys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.infrastructure.time.StubTimeProvider;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.TestSpecFactory;
+
+class VoluntaryExitDataProviderTest {
+  private final Spec spec = TestSpecFactory.createMinimal(SpecMilestone.CAPELLA);
+  private final StubTimeProvider timeProvider = StubTimeProvider.withTimeInSeconds(100000);
+  private VoluntaryExitDataProvider provider;
+
+  @BeforeEach
+  void setUp() {
+    provider = new VoluntaryExitDataProvider(spec, timeProvider);
+  }
+
+  @Test
+  void calculateExpectedEpochWhenNotProvided() {
```

Comment:
We probably need to update the name of this method as we don't have the optional parameter anymore.